Src/AutoMoq/MockType.cs

```diff
@@ -68,11 +68,14 @@ internal static Type GetMockedType(this Type type)
             where TMock : class
         {
             return setup.Returns(() =>
-                {
-                    var result = (TResult) context.Resolve(typeof (TResult));
-                    setup.Returns(result);
-                    return result;
-                });
+            {
+                var specimen = context.Resolve(typeof (TResult));
+                TResult result = specimen is TResult ? (TResult) specimen : default(TResult);

```

Comment:
Well, as far as I can tell, we have 3 options:

* Leave it as it is, and throw an `InvalidCastException` when an `OmitSpecimen` is found
* Return `null`
* Throw an `ObjectCreationException`.

As you already pointed out, returning `null` is not desirable. Option 1 is clearly not desirable either.

The third option would be, IMO, appropriate. In the context of setting up a method, an `OmitSpecimen` does not make sense. There's nothing to omit. So we need to tell the user "hey, we tried to setup this method to return an instance of `string`, but we couldn't create one".